rust/src/countme/repo.rs

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+use anyhow::{Context, Result};
+use std::fs;
+use std::path::PathBuf;
+
+use ini::Ini;
+
+/// Location for DNF repositories configuration
+pub const YUM_REPOS_D: &str = "/etc/yum.repos.d";
+
+/// Repository configuration
+/// Not exhaustive and only includes the options needed for Count Me support.
+#[derive(Debug)]
+pub struct Repo {
+    name: String,
+    enabled: bool,
+    count_me: bool,
+    meta_link: String,
+}
+
+/// From https://github.com/rpm-software-management/libdnf/blob/45981d5f53980dac362900df65bcb2652aa8d7c7/libdnf/conf/OptionBool.hpp#L30-L31
+fn is_true(string: &str) -> bool {
+    string == "1" || string == "yes" || string == "true" || string == "on"
+}
+
+/// Read all repository configuration files from the default location
+pub fn all() -> Result<Vec<Repo>> {
+    let configs = fs::read_dir(YUM_REPOS_D)
+        .with_context(|| format!("Could not list files in: {}", YUM_REPOS_D))?;
+    let mut repos = Vec::new();
+    for c in configs {
+        let path = match c {
+            Err(e) => {
+                println!("Could not list file in : {}: {}", YUM_REPOS_D, e);
+                continue;
+            }
+            Ok(p) => p.path(),
+        };
```

Comment:
Let's just error out in this case, like
```suggestion
        let path = c?.path();
```
or so.